config/routes.rb

```diff
@@ -1,6 +1,12 @@
 Rails.application.routes.draw do
+  devise_for :users
   # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
 
   # Defines the root path route ("/")
-  # root "articles#index"
+   root "splash#index"
+  resources :main, only: %i[index]
+  resources :users
+  resources :categories, only: %i[index new create update show destroy]  do
+  resources :expenses , only: %i[index new create destroy]
```

Comment:
- [ ] As requested, please add the 'update' action here as you have it in your controller, or remove it.